### DISCUSSION TITLE
"brew livecheck" - updating a homebrew formula in my tap

### DISCUSSION THREAD

**Discussion by abitrolly** [Homebrew/discussions#1934] [Getting started]
Is there a way to quickly bump version in my tap, which is `abitrolly/tap/tdu`?

I've just noticed the `brew livecheck` command that somehow managed to see that there is a new version of my formula.
```bash
$ brew livecheck tdu      
tdu : 1.34 ==> 1.36
```

There is a `brew bump` command that shows more or less same info, but it it does bump anything.
```bash
$ brew bump tdu
==> tdu
Current formula version:  1.34
Latest Repology version:  not found
Latest livecheck version: 1.36
Open pull requests:       none
```

But it doesn't really bump anything. I expected some `brew bump tdu --commit` magic there to bump and commit version with new SHA in my repository, but it is not implemented.

I searced for "brew livecheck updating a homebrew formula" but didn't get any cheat sheets explaining.

So what people use to bump their formulas in the most automated way?

**Comment by carlocab** [comment]
Try `brew bump-formula-pr`. You might also be interested in https://github.com/dawidd6/action-homebrew-bump-formula.

**Reply by abitrolly** [reply]
Doesn't work out of the box. :eyes: 

```bash
$ brew bump-formula-pr tdu
...
Error: Invalid usage: tdu: no --url= or --version= argument specified!
```

**Reply by abitrolly** [reply]
@carlocab the GitHub Action is nice, but I was thinking for something I can still invoke manually from CLI. I don't mind using  `brew bump-formula-pr` but I need to figure out how to get URL for formula to pass to it.

**Reply by SMillerDev** [reply]
It's the URL for the file to download.

**Reply by abitrolly** [reply]
@SMillerDev I found that `brew bump-formula-pr tdu --version=1.36` could find the right file automatically. Then I found a way to extract the new detected version from "livecheck".
```
$ brew livecheck --json tdu | grep -oP '(?<=latest": ")[.\d]+'
1.36
```
Now I need to figure out how to combine that into a single command, so that it can be invoked like like `burp tdu`.

**Reply by abitrolly** [reply]
So I came up with this onliner.

```bash
(export F=tdu; export LATEST=$(brew livecheck --json tdu | grep -oP '(?<=latest": ")[.\d]+'); echo "bumping $F to $LATEST"; brew bump-formula-pr $F --version=$LATEST --no-fork)
```
The problem is that I need to edit `F=tdu` for any other package. Another problem is that it tries to create PR itself. I would prefer for it to just open a browser page and let me confirm that. Instead `bump-formula-pr` fails with an error and requires me to manualy click the link.

```
...
remote: 
remote: Create a pull request for 'bump-tdu-1.36' on GitHub by visiting:
remote:      https://github.com/abitrolly/homebrew-tap/pull/new/bump-tdu-1.36
remote: 
To github.com:abitrolly/homebrew-tap.git
 * [new branch]      bump-tdu-1.36 -> bump-tdu-1.36
Branch 'bump-tdu-1.36' set up to track remote branch 'bump-tdu-1.36' from 'git@github.com:abitrolly/homebrew-tap.git'.
Error: Unable to open pull request: No GitHub credentials found in macOS Keychain or environment.
Create a GitHub personal access token:
    https://github.com/settings/tokens/new?scopes=gist,public_repo,workflow&description=Homebrew
  echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.zshrc
!
```
